### PR TITLE
EZP-28926: Add validation for PermissionResolver::hasAccess(module, function) arguments

### PR DESCRIPTION
**Pull request by kmadejski**
| Question           | Answer
| ------------------ | ------------------
| **JIRA issue**     | [EZP-28926](https://jira.ez.no/browse/EZP-28926)
| **Improvement**| yes
| **New feature**    | no
| **Target version** | `master`
| **BC breaks**      | yes*
| **Tests pass**     | yes/no
| **Doc needed**     | no

As described in the JIRA issue, we need to add validation for `PermissionResolver::hasAccess(module, function)` arguments. 

Moreover, `eZ/Bundle/EzPublishCoreBundle/Resources/config/policies.yml` was moved to the `eZ\Core\settings\policies.yml`. The same policies are used by integration tests and `EzPublishCoreBundle`.

*- Public API method will throw a new `\eZ\Publish\API\Repository\Exceptions\InvalidArgumentException` so it might be considered as a BC break.

**Question**:
~~Is it safe to completely remove `eZ\Bundle\EzPublishCoreBundle\DependencyInjection\Security\PolicyProvider\RepositoryPolicyProvider` or might it be extended somewhere?~~ Marked as deprecated.

**TODO**:
- [x] Implement improvement.
- [x] Implement tests.
- [x] Fix new code according to Coding Standards (`$ composer fix-cs`).
- [x] Ask for Code Review.
- [x] Improve test code quality*

*- as discussed with @alongosz we can change every `array()` to short syntax version in test code, at least in data providers. It will be done in the separated commit later.

### CODE REVIEW

File: eZ/Publish/Core/Repository/RoleService.php [hunk 1]
```diff
@@ -1410,7 +1382,6 @@ protected function validatePolicy($module, $function, array $limitations)
                         "'{$limitation->getIdentifier()}' was found several times among the limitations"
                     );
                 }
-
```
Comment (andrerom) [inline eZ/Publish/Core/Repository/RoleService.php]: can keep this line

Comment (kmadejski) [reply eZ/Publish/Core/Repository/RoleService.php]: Sorry, it was deleted by mistake. I added it back 🙂

**Comment by andrerom** [comment]
> Improve test code quality*

@kmadejski Has this been done yet or still pending? _(this is why we usually don't review before everything is done to avoid confusions for people merging. and to not have to add additional reviews of changes done after review ;) )_

**Comment by kmadejski** [comment]
> Has this been done yet or still pending?

This is now done. 

> (this is btw why we usually don't review before everything is done to avoid confusions for people merging. and to not have to add/do additional reviews of changes done after review ;) )

I see the point now. I'll avoid this kind of mess next time 🙂

**Comment by andrerom** [comment]
Ok, seems to be about to pass. Are you ok as well with this now @alongosz ?

**Comment by emodric** [comment]
FYI: This breaks checking for permissions used by eZ Publish Legacy but undefined in the new stack, for example `setup/managecache` or similar. Is this BC break really required/needed?

**Comment by andrerom** [comment]
>  Is this BC break really required/needed?
We had several issues where code was using wrong modules / policies, this made us found a few more such issues. So imo; yes it is.

Maybe you should inject those policies so they are treated as valid?

**Comment by emodric** [comment]
> Maybe you should inject those policies so they are treated as valid?

That was my first idea, but that wouldn't help with 3rd party extensions. Many of those define their own modules and functions and injecting all of those is not feasible :)

I might have a solution for legacy bridge though. Stay tuned :)

**Pull request merged**